src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTrigger.java

```diff
@@ -142,7 +145,7 @@ private StandardUsernamePasswordCredentials getCredentials() {
         CredentialsProvider.lookupCredentials(
             StandardUsernamePasswordCredentials.class,
             this.job,
-            Tasks.getDefaultAuthenticationOf(this.job),
+            Tasks.getDefaultAuthenticationOf((ParameterizedJob) job),
```

Comment:
no guard here, how can we be sure it's a ParametrizedJob?

Reply:
> no guard here, how can we be sure it's a ParametrizedJob?

We have `isApplicable` which should filter out jobs without the `ParameterizedJob` interface. I can add an extra check here, but we need to be very careful here. The code assumes that there are valid credentials and tries to authenticate with Stash, even if the userspace and the password are empty.

We should implement a way to communicate to the trigger that the job is not properly configured and should not be run.

Reply:
For now just adding a code comment explaining why we expect it to be `ParameterizedJob` is good enough for me

Reply:
I've added comment to both casts. The casts are on separate lines now to improve readability.